---
title: C6255
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C6255
helpviewer_keywords:
- C6255
ms.assetid: bb6430b2-782a-4410-a8e1-609df06007de
author: mikeblome
ms.author: mblome
manager: markl
ms.workload:
- multiple
ms.openlocfilehash: ca09619a9a7258d24f9f60ad8d6fa27553123709
ms.sourcegitcommit: 174c992ecdc868ecbf7d3cee654bbc2855aeb67d
ms.translationtype: MT
ms.contentlocale: pt-BR
ms.lasthandoff: 12/06/2019
ms.locfileid: "74878561"
---
# <a name="c6255"></a>C6255
aviso C6255: _alloca indica a falha gerando uma exceção de estouro de pilha. Considere utilizar _malloca no lugar

 Este aviso indica que uma chamada a `_alloca` foi detectada fora do tratamento local de exceção. `_alloca` deve sempre ser chamado de dentro do intervalo protegido de um manipulador de exceção, porque pode lançar uma exceção de estouro de pilha em caso de falha. Se possível, em vez de usar `_alloca`, considere usar `_malloca` que é uma versão mais segura de `_alloca`.

## <a name="example"></a>Exemplo
 O código a seguir gerencia este aviso, pois `_alloca` pode gerar uma exceção:

```cpp
#include <windows.h>

void f( )
{
  void *p = _alloca(10);
  // code ...
}
```

 Para corrigir este aviso, use `_malloca` e adicione o manipulador de exceção, conforme mostrado no código o seguir:

```cpp
#include <windows.h>
#include <malloc.h>

void f( )
{
  void *p;
  int errcode;
  __try
  {
    p = _malloca(10);
    // code...
    _freea(p);
  }
  __except( (GetExceptionCode() == STATUS_STACK_OVERFLOW ) ?
              EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH )
  {
    errcode = _resetstkoflw();
    // code ...
  }
}
```

## <a name="see-also"></a>Consulte também
 [_malloca](/cpp/c-runtime-library/reference/malloca)
